src/main/java/internet/shop/dao/interfaces/ProductDao.java

```diff
@@ -0,0 +1,19 @@
+package internet.shop.dao.interfaces;
+
+import internet.shop.models.Product;
+import java.util.List;
+import java.util.Optional;
+
+public interface ProductDao {
+    Product create(Product product);
+
+    Optional<Product> getById(Long productId);
+
+    Product update(Product product);
+
+    boolean deleteById(Long productId);
+
+    boolean delete(Product product);
+
+    List<Product> getAllProducts();
```

Comment:
here you should have the following methods:

- Product create(Product product);
- Optional<Product> get(Long id);
- List<Product> getAll();
- Product update(Product product);
- boolean delete(Long id);